R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -101,45 +102,47 @@ module_socioeconomics_L100.Population_downscale_ctry <- function(command, ...) {
       replace_na(list(pop = 0)) %>%
       group_by(year) %>%
       summarize(pop = sum(pop))
+
     # Subtract reported countries from total global - these are the total population values in each period that will be allocated to countries with missing values
     pop_missing <- left_join(pop_global, pop_notmissing, by = "year") %>%
       mutate(pop_allocate = pop.x - pop.y) %>%
-      select(year, pop_allocate) %>%
-      mutate(year = paste0("X", year)) %>%
-      spread(year, pop_allocate) # The next set of calculations are much easier to do in wide format.
+      select(year, pop_allocate)
+
+    # Fifth, estimate historic population values for countries with missing values and calculate ratios relative to last historical year (1950)
+    maddison_hist_ratio <-
+      filter(hist_interp, iso != "world_total") %>%
+      left_join(pop_missing, by = "year")
+
+    maddison_hist_years <- unique(maddison_hist_ratio$year)
+
+    # Fill in missing values for period t using ratio of country to total population in period t+1 times the missing share in period t.
+    for( i in rev(maddison_hist_years)[-1]){
+      maddison_hist_ratio %>%
+        ungroup() %>%
+        mutate(pop2 = if_else(is.na(pop), (lead(pop, n = 1L, order_by = iso)), 0)) %>%
```

Comment:
i know this is my fault. but can we make this better? 
i think pop2 should be previous_pop/
and else should probably be NA. 
0 and pmax was a hack.